src/Sylius/Component/Core/Promotion/Action/UnitFixedDiscountPromotionActionCommand.php

```diff
@@ -80,4 +87,31 @@ private function setUnitsAdjustments(OrderItemInterface $item, int $amount, Prom
             );
         }
     }
+
+    /**
+     * @param array<string, mixed> $configuration
+     * @return array<string, mixed>
+     */
+    private function filterItems(OrderInterface $subject, array $configuration, string $channelCode): array
+    {
+        $filteredItems = [];
+
+        if (null === $this->compositeFilter) {
+            $filteredItems = $this->priceRangeFilter->filter(
+                $subject->getItems()->toArray(),
+                array_merge(['channel' => $subject->getChannel()], $configuration[$channelCode]),
+            );
+            $filteredItems = $this->taxonFilter->filter($filteredItems, $configuration[$channelCode]);
+            $filteredItems = $this->productFilter->filter($filteredItems, $configuration[$channelCode]);
+        }
+
+        if (null !== $this->compositeFilter) {
+            $filteredItems = $this->compositeFilter->filter(
+                $subject->getItems()->toArray(),
+                array_merge(['channel' => $subject->getChannel()], $configuration[$channelCode]),
+            );
+        }
```

Comment:
I've a feeling, that ECS will suggest this change anyway, so it can be done now.

```suggestion
        if (null === $this->compositeFilter) {
            $filteredItems = $this->priceRangeFilter->filter(
                $subject->getItems()->toArray(),
                array_merge(['channel' => $subject->getChannel()], $configuration[$channelCode]),
            );
            $filteredItems = $this->taxonFilter->filter($filteredItems, $configuration[$channelCode]);
            $filteredItems = $this->productFilter->filter($filteredItems, $configuration[$channelCode]);
            
            return $filteredItems;
        }

        $filteredItems = $this->compositeFilter->filter(
            $subject->getItems()->toArray(),
            array_merge(['channel' => $subject->getChannel()], $configuration[$channelCode]),
        );
```